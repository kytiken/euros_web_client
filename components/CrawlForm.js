import React from 'react';
import PropTypes from 'prop-types';
import config from '../config';

class CrawlForm extends React.Component {
  constructor(props) {
    super(props);
    this.urlInput = null;
    this.depthLimitInput = null;
    this.timeoutInput = null;
    this.recvTimeoutInput = null;
    this.patternInput = null;
    this.cookieInput = null;
  }

  render() {
    return (
      <div>
        <div>
          <label htmlFor="urlInput">
            url:
            <input ref={(input) => { this.urlInput = input; }} id="urlInput" type="text" />
          </label>
        </div>
        <div>
          <label htmlFor="depthLimitInput">
            depth_limit:
            <input ref={(input) => { this.depthLimitInput = input; }} id="depthLimitInput" type="text" />
          </label>
        </div>
        <div>
          <label htmlFor="timeoutInput">
            timeout:
            <input ref={(input) => { this.timeoutInput = input; }} id="timeoutInput" type="text" />
          </label>
        </div>
        <div>
          <label htmlFor="recvTimeoutInput">
            recv_timeout:
            <input ref={(input) => { this.recvTimeoutInput = input; }} id="recvTimeoutInput" type="text" />
          </label>
        </div>
        <div>
          <label htmlFor="patternInput">
            pattern:
            <input ref={(input) => { this.patternInput = input; }} id="patternInput" type="text"  />
          </label>
        </div>
        <div>
          <label htmlFor="cookieInput">
            cookie:
            <input ref={(input) => { this.cookieInput = input; }} id="cookieInput" type="text" />
          </label>
        </div>
        <button onClick={() => {
          const crawl = {
            url: this.urlInput.value,
            depth_limit: this.depthLimitInput.value,
            timeout: this.timeoutInput.value,
            recv_timeout: this.recvTimeoutInput.value,
            pattern: this.patternInput.value,
            cookie: this.cookieInput.value,
          };
          fetch(config.crawl_url, {
            method: 'POST',
            body: JSON.stringify({ crawl }),
            headers: new Headers({
              'Content-Type': 'application/json',
              Accept: 'application/json',
            }),
          })
          .then(res => res.json())
          .then(response => this.props.addCrawl(response.data))
          .catch(error => console.error('Error:', error));
        }}>
        submit
        </button>
      </div>
    );
  }
}

CrawlForm.propTypes = {
  addCrawl: PropTypes.func.isRequired,
};

export default CrawlForm;
