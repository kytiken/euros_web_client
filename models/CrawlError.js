import { Record } from 'immutable';

const crawlErrorAttr = {
  url: null,
  depthLimit: null,
  timeout: null,
  recvTimeout: null,
  pattern: null,
  cookie: null,
};

export default class CrawlError extends Record(crawlErrorAttr) {
  isRaised() {
    return this.url ||
      this.depthLimit ||
      this.timeout ||
      this.recvTimeout ||
      this.pattern ||
      this.cookie;
  }
}
