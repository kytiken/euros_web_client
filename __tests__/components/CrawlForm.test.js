import React from 'react';
import renderer from 'react-test-renderer';
import CrawlForm from '../../components/CrawlForm';
import CrawlError from '../../models/CrawlError';

it('renders correctly', () => {
  const tree = renderer
    .create(<CrawlForm errors={new CrawlError()} createCrawl={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
