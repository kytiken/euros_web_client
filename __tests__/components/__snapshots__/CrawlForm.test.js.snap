// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div>
      <label
        htmlFor="urlInput"
      >
        url:
        <input
          id="urlInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <div>
      <label
        htmlFor="depthLimitInput"
      >
        depth_limit:
        <input
          id="depthLimitInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <div>
      <label
        htmlFor="timeoutInput"
      >
        timeout:
        <input
          id="timeoutInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <div>
      <label
        htmlFor="recvTimeoutInput"
      >
        recv_timeout:
        <input
          id="recvTimeoutInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <div>
      <label
        htmlFor="patternInput"
      >
        pattern:
        <input
          id="patternInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <div>
      <label
        htmlFor="cookieInput"
      >
        cookie:
        <input
          id="cookieInput"
          type="text"
        />
      </label>
      <span />
    </div>
    <button
      onClick={[Function]}
    >
      submit
    </button>
  </div>
</div>
`;
